applications/RomApplication/costum_strategies/rom_builder_and_solver.h

```diff
@@ -256,45 +272,66 @@ class ROMBuilderAndSolver : public BuilderAndSolver<TSparseSpace, TDenseSpace, T
         }
     }
 
-    void ProjectToReducedBasis(
-        const TSystemVectorType &rX,
-        ModelPart::NodesContainerType &rNodes,
-        Vector rom_unknowns)
-    {
-        for (const auto &node : rNodes){
-            unsigned int node_aux_id = node.GetValue(AUX_ID);
-            const auto &nodal_rom_basis = node.GetValue(ROM_BASIS);
-            for (int i = 0; i < mRomDofs; ++i){
-                for (int j = 0; j < mNodalDofs; ++j){
-                    rom_unknowns[i] += nodal_rom_basis(j, i) * rX(node_aux_id * mNodalDofs + j);
-                }
-            }
-        }
-    }
+
+    // Vector ProjectToReducedBasis(
+	// 	const TSystemVectorType& rX,
+	// 	ModelPart::NodesContainerType& rNodes
+	// )
+    // {
+    //     Vector rom_unknowns = ZeroVector(mRomDofs);
+    //     for(const auto& node : rNodes)
+    //     {
+    //         unsigned int node_aux_id = node.GetValue(AUX_ID);
+    //         const auto& nodal_rom_basis = node.GetValue(ROM_BASIS);
+	// 			for (int i = 0; i < mRomDofs; ++i) {
+	// 				for (int j = 0; j < mNodalDofs; ++j) {
+	// 					rom_unknowns[i] += nodal_rom_basis(j, i)*rX(node_aux_id*mNodalDofs + j);
+	// 				}
+	// 			}
+    //     }
+    //     return rom_unknowns;
+	// }
 
     void ProjectToFineBasis(
         const TSystemVectorType &rRomUnkowns,
-        ModelPart::NodesContainerType &rNodes,
-        TSystemVectorType &rX)
+        ModelPart &rModelPart,
+        TSystemVectorType &Dx)
     {
-        TSparseSpace::SetToZero(rX);
-        for (const auto &node : rNodes){
-            unsigned int node_aux_id = node.GetValue(AUX_ID);
-            const auto &nodal_rom_basis = node.GetValue(ROM_BASIS);
-            Vector tmp = prod(nodal_rom_basis, rRomUnkowns);
-            for (unsigned int i = 0; i < tmp.size(); ++i){
-                rX[node_aux_id * mNodalDofs + i] = tmp[i];
-            }
+        const Matrix *current_rom_nodal_basis{nullptr};

```

Comment:
mmm, it would be cool to find a way of doing this in parallel (it is a time consuming loop)

leave the change for a future PR, but please do consider it

Reply:
I am doing the OpenMP parallelization in a separate PR, as soon as this one gets aproved...